segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/hdfs/HDFSStorage.java

```diff
@@ -639,9 +639,6 @@ private int readInternal(SegmentHandle handle, byte[] buffer, long offset, int b
         //There is only one file per segment.
         FileStatus currentFile = findStatusForSegment(handle.getSegmentName(), true);
         try (FSDataInputStream stream = this.fileSystem.open(currentFile.getPath())) {
-            if (offset + length > stream.available()) {
```

Comment:
From this removed code, it looks like `FSDataInputStream.available()` was incorrectly interpreted as the total amount of data available in the file, and it isn't according to this documentation:

https://docs.oracle.com/javase/8/docs/api/java/io/FilterInputStream.html?is-external=true#available--

I think removing is the right thing to do.